compiler/cpp/src/thrift/generate/t_java_generator.cc

```diff
@@ -1060,11 +1060,15 @@ void t_java_generator::generate_union_getters_and_setters(ostream& out, t_struct
     }
     indent(out) << "public void set" << get_cap_name(field->get_name()) << "("
                 << type_name(field->get_type()) << " value) {" << endl;
+
+    indent(out) << "  setField_ = _Fields." << constant_name(field->get_name()) << ";" << endl;
+
     if (type_can_be_null(field->get_type())) {
-      indent(out) << "  if (value == null) throw new java.lang.NullPointerException();" << endl;
+      indent(out) << "  value_ = java.util.Objects.requireNonNull(value);" << endl;
```

Comment:
This is nice, but the method can take an optional argument to have a string message. It might be good to put the name of the field in the message:

```suggestion
      indent(out) << "  value_ = java.util.Objects.requireNonNull(value,\"" << "_Fields." << constant_name(field->get_name()) << "\");" << endl;
```

(or similar)

Reply:
Also, this will only work with Java 7 or later. I'm not sure what version of Java libthrift is targeting. Do we know it's definitely for 7 or later? I'm unable to confirm using my normal method (inspecting the pom.xml) because the pom.xml is bolted on afterwards, and libthrift isn't built with Maven (AFAICT).

Reply:
My expectation is that Thirft is running on JDK 1.8 or higher.

https://issues.apache.org/jira/browse/THRIFT-4694

Reply:
Perfect. Thanks.